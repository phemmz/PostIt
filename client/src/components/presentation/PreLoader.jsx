import React from 'react';

/**
 * @description PreLoader Component
 * @return {*} div
 */
const PreLoader = () => {
  return (
    <div className="container">
      <div className="col s3 offset-s4 my-preloader">
        <div className="valign-wrapper preloader-wrapper active">
          <div className="spinner-layer spinner-red-only">
            <div className="circle-clipper left">
              <div className="circle" />
            </div>
            <div className="gap-patch">
              <div className="circle" />
            </div>
            <div className="circle-clipper right">
              <div className="circle" />
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default PreLoader;
