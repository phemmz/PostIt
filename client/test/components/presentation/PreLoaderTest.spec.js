import React from 'react';
import { shallow } from 'enzyme';
import PreLoader from '../../../src/components/presentation/PreLoader';

describe('PreLoader Component', () => {
  it('should render a materialize preloader', () => {
    const wrapper = shallow(<PreLoader />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div').exists()).toBe(true);
    expect(wrapper.find('div').length).toBe(10);
  });
});
