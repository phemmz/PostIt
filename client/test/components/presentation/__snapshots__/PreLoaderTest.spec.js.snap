// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreLoader Component should render a materialize preloader 1`] = `
<div
  className="container"
>
  <div
    className="col s3 offset-s4 my-preloader"
  >
    <div
      className="valign-wrapper preloader-wrapper active"
    >
      <div
        className="spinner-layer spinner-red-only"
      >
        <div
          className="circle-clipper left"
        >
          <div
            className="circle"
          />
        </div>
        <div
          className="gap-patch"
        >
          <div
            className="circle"
          />
        </div>
        <div
          className="circle-clipper right"
        >
          <div
            className="circle"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
